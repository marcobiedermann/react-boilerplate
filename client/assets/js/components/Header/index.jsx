import PropTypes from 'prop-types';
import React from 'react';

const Header = props => {
  const { children } = props;

  return <header>{children}</header>;
};

Header.propTypes = {
  children: PropTypes.node,
};

Header.defaultProps = {
  children: null,
};

export default Header;
