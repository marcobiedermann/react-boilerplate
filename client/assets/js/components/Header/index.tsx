import React from 'react';

const Header: React.FC = (props) => {
  const { children } = props;

  return <header>{children}</header>;
};

export default Header;
