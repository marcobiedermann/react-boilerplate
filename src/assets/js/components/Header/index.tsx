import React, { ReactNode } from 'react';

export interface HeaderProps {
  children: ReactNode;
}

function Header(props: HeaderProps): JSX.Element {
  return <header {...props} />;
}

export default Header;
