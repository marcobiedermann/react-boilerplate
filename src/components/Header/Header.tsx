import { ReactNode } from 'react';

interface HeaderProps {
  children: ReactNode;
}

function Header(props: HeaderProps): JSX.Element {
  return <header {...props} />;
}

export type { HeaderProps };
export default Header;
