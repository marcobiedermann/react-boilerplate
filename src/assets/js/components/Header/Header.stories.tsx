import { ComponentMeta, ComponentStoryObj } from '@storybook/react';
import Header from '.';

export default {
  component: Header,
} as ComponentMeta<typeof Header>;

export const Default: ComponentStoryObj<typeof Header> = {
  args: {
    children: 'Header',
  },
};
