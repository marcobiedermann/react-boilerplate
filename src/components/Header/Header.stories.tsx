import type { Meta, StoryObj } from '@storybook/react';
import Header from '.';

const meta: Meta<typeof Header> = {
  component: Header,
};

type Story = StoryObj<typeof Header>;

const Default: Story = {
  args: {
    children: 'Header',
  },
};

export { Default };
export default meta;
