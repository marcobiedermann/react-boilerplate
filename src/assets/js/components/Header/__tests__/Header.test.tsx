/* eslint-disable import/no-extraneous-dependencies */

import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import * as stories from '../Header.stories';

const { Default } = composeStories(stories);

describe('header', () => {
  it('should render correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
