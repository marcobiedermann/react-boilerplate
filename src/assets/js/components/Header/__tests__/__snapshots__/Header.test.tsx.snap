// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should render correctly 1`] = `
<header>
  Header
</header>
`;
